--- a/advance/menu/mm.cc
+++ b/advance/menu/mm.cc
@@ -651,47 +651,54 @@
 }
 
 //---------------------------------------------------------------------------
-// Lectura de archivos amp - Layouts
-
-bool layouts_load(config_state& rs)
-{
+// Lectura de archivos Layouts
+
+bool layout_read(config_state& rs, const string& custom_path)
+{	
+	if (custom_path == "")
+		return true;
+		
+	string path_custom_completo = file_config_file_home(custom_path.c_str());
+	string dir_custom = file_dir_custom(path_custom_completo); 
+		
+	load_current_dir_custom(dir_custom.c_str());
+
+	if (access(path_custom_completo.c_str(), F_OK)!=0) {
+		target_err("Error opening file '%s'\n", path_custom_completo.c_str());
+		return true;
+	}
+	
 	adv_conf* custom_context;
 	const char* section_map_custom[1];
+	custom_context = conf_init();
+	config_state::conf_register_custom(custom_context);
+	section_map_custom[0] = (char*)"";
+	conf_section_set(custom_context, section_map_custom, 1);
+
+	if (conf_input_file_load_adv(custom_context, 3, path_custom_completo.c_str(), 0, 0, 1, STANDARD, sizeof(STANDARD)/sizeof(STANDARD[0]), error_callback, 0) != 0) {
+		conf_done(custom_context);
+		return false;
+	}
+
+	if (!rs.load_custom(custom_context, custom_path)) {
+		conf_done(custom_context);
+		return false;
+	}
+
+	conf_done(custom_context);
+
+	return true;
+}
+
+bool layouts_load(config_state& rs)
+{
+	if (!layout_read(rs, rs.menu_systems->custom_file_path_get()))
+		return false;
 	
-	for(pemulator_container::iterator j = rs.emu.begin(); j!=rs.emu.end(); j++) {
-
-		string path_archivo_custom = (*j)->custom_file_path_get();
-		string nombre_emulador = (*j)->user_name_get();
-
-		if (path_archivo_custom != "") {
-			string path_custom_completo = file_config_file_home(path_archivo_custom.c_str());
-			
-			string dir_custom = file_dir_custom(path_custom_completo); 
-			load_current_dir_custom(dir_custom.c_str());
-			
-			if (access(path_custom_completo.c_str(), F_OK)==0) {
-				custom_context = conf_init();
-				config_state::conf_register_custom(custom_context);
-				section_map_custom[0] = (char*)"";
-				conf_section_set(custom_context, section_map_custom, 1);
-
-				if (conf_input_file_load_adv(custom_context, 3, path_custom_completo.c_str(), 0, 0, 1, STANDARD, sizeof(STANDARD)/sizeof(STANDARD[0]), error_callback, 0) != 0) {
-					conf_done(custom_context);
-					return false;
-				}
-
-				if (!rs.load_custom(custom_context, path_archivo_custom)) {
-					conf_done(custom_context);
-					return false;
-				}
-
-				conf_done(custom_context);
-
-			} else {
-				target_err("Error opening file '%s'\n", path_custom_completo.c_str());
-			}
-		}
-	}
+	for(pemulator_container::iterator j = rs.emu.begin(); j!=rs.emu.end(); j++)
+		if (!layout_read(rs, (*j)->custom_file_path_get()))
+			return false;
+	
 	return true;
 }